src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -3237,7 +3240,7 @@ void Compiler::fgDebugCheckNodeLinks(BasicBlock* block, Statement* stmt)
         // TODO: return?
     }
 
-    assert(fgStmtListThreaded);
+    assert(fgNodeThreading != NodeThreading::None);
```

Comment:
I would just consider checking the new locals threading for now, since omissions from the all node threading more likely have very observable consequences (I know I've gotten asserts when I forgot to resequence trees).

Reply:
Done, I added a new `PhaseChecks::CHECK_LINKED_LOCALS` as separating out the parts I wanted from `CHECK_FG` turned out to be a larger refactoring that I don't think should be part of this PR.